### ISSUE TITLE
SpeechServices - Error: ConnectionFailure 

### ISSUE THREAD

**Issue by laultman** [Azure-Samples/cognitive-services-speech-sdk#1723]
**Describe the bug**
**Fixed** - see comments below...
Trying to run the get started text to speech sample code. I have an established connection to Azure Government and have an active F0 account setup for testing. I have the service key for the SpeechServices, the region ID from the portal, and verified these in my local machine environment variables. I get the following error:
Type some text that you want to speak...
> help
CANCELED: Reason=Error
CANCELED: ErrorCode=ConnectionFailure
CANCELED: ErrorDetails=[Connection failed (no connection to the remote host). Internal error: 1. Error details: Failed with error: WS_OPEN_ERROR_UNDERLYING_IO_OPEN_FAILED
wss://usgovvirginia.tts.speech.microsoft.com/cognitiveservices/websocket/v1
X-ConnectionId: 0358c3c9b38541b99305d8f4cf664042 USP state: 2. Received audio size: 0 bytes.]
CANCELED: Did you update the subscription info?

**To Reproduce**
Following the GetStarted Text to Speech documentation. Downloaded the samples from GitHub.  Opened the sample dotnetcore solution file in VS 2022 17.3.6. With a program debug break, I verified the key and region are valid. Continued the run. Get error.
I then upgraded the project to .NET6.0 and updated the Microsoft Speech SDK to 1.24.1. The error still exists.

**Expected behavior**
Since this is a simple get started, no frills, test, I expected it to connect to Azure service and run.

**Version of the Cognitive Services Speech SDK**
Which version of the SDK are you using. 1.24.0 then updated to 1.24.1

**Platform, Operating System, and Programming Language**
Windows 11 (fully patched, latest updates) Visual Studio 2022 17.3.6, C#, .net6.0

**Additional context**
 - Error messages, stack trace, sample code to reproduce the issue, ...
 - Please provide full Speech SDK Log, as per the instructions [here](https://docs.microsoft.com/azure/cognitive-services/speech-service/how-to-use-logging)?
 - If you report an error for a specific service interaction, please report the SessionId and time (incl. timezone) of the reported incidents. The SessionId is reported in all call-backs/events you receive.
 - Any other additional information

**Comment by laultman** [comment]
Troubleshooting using PowerShell: The following command with my valid key, successfully returned the connection token.
Note: speechtest is the generated custom domain for the service. I am using the All connections however to avoid any firewall issues.

$FetchTokenHeader = @{
  'Content-type'='application/x-www-form-urlencoded'
  'Content-Length'= '0'
  'Ocp-Apim-Subscription-Key' = 'my key here'
}
$OAuthToken = Invoke-RestMethod -Method POST -Uri https://speechtest.cognitiveservices.azure.us/sts/v1.0/issuetoken -Headers $FetchTokenHeader
$OAuthToken

**Comment by laultman** [comment]
Tried again using a different config for the SpeechConfig class.
            var config = SpeechConfig.FromEndpoint(new Uri("https://speechtest.cognitiveservices.azure.us/sts/v1.0/issuetoken"), speechKey);

Same problem different error:
CANCELED: Reason=Error
CANCELED: ErrorCode=RuntimeError
CANCELED: ErrorDetails=[Internal error: HttpAPI failed - Failed with error: HTTPAPI_OPEN_REQUEST_FAILED [0x3 | 11001]
POST https://speechtest.tts.speech.microsoft.com/cognitiveservices/v1
X-ConnectionId: fc934fab75b84187b353a061295ef1eb]

**Comment by laultman** [comment]
Following this, [documentation ](https://learn.microsoft.com/en-us/azure/cognitive-services/speech-service/sovereign-clouds?tabs=c-sharp)I got it to work.
The code for the SpeechConfig class:
`var config = SpeechConfig.FromHost(new Uri(speechRegionGov), speechKey);`

The uri for the speechRegionGov is found in a table on the linked document. I added this to my environment variables along with my service key. If your VS environment is running when you add/edit your environment variables, you must restart VS or they will be ignored.
